api/integrations/launch_darkly/launch_darkly.py

```diff
@@ -0,0 +1,239 @@
+        ld_environments = self.client.get_environments(ld_project_id)
+        if not ld_environments:
+            # todo do we want to stop the script if no environments can be found
+            self.logger.warning(
+                f"No environments found in Launch Darkly for project id: {ld_project_id}"
+            )
+
+        self.logger.info(
+            f"Attempting to retrieve Launch Darkly flags under project id: {ld_project_id}"
+        )
+        ld_flags = self.client.get_flags(ld_project_id)
+        if not ld_flags:
+            # todo do we want to stop the script if no flags can be found
+            self.logger.warning(
+                f"No flags found in Launch Darkly for project id: {ld_project_id}"
+            )
+
+        environments = []
+        self.logger.info(f"Adding {len(ld_environments)} environments")
+        for ld_environment in ld_environments:
+            environment = self._create_environment(ld_environment, project)
+            environments.append(environment)
+
+        self.logger.info(f"Adding {len(ld_flags)} features")
+        for ld_flag in ld_flags:
+            self._create_feature(ld_flag, environments, project)
+
+        self.logger.completed_at(datetime.now())
+        self.logger.completed = True
+        self.logger.save()
+
+        self.logger.info("Finished importing")
+
+    def _create_project(self, organisation_id: UUID, ld_project: dict) -> Project:
+        return Project.objects.create(
+            organisation_id=organisation_id, name=ld_project.get("name")
+        )
+
+    def _create_environment(self, ld_environment, project: Project) -> Environment:
+        return Environment.objects.create(
+            name=ld_environment.get("name"),
+            project=project,
+        )
+
+    def _create_feature(
+        self, ld_flag, environments: list[Environment], project: Project
+    ):
+        match ld_flag.get("kind"):
+            case "boolean":
+                self._create_boolean_feature(ld_flag, environments, project)
+            case "multivariate":
+                self._create_multivariate_feature(ld_flag, environments, project)
+            case _:
+                raise Exception("Invalid flag type from Launch Darkly")
+
+    def _create_boolean_feature(
+        self, ld_flag, environments: list[Environment], project: Project
+    ):
+        # todo try get tags
+        try:
+            feature = Feature.objects.create(
+                name=ld_flag.get("key"),
+                project=project,
+                initial_value=None,  # todo
+                description=ld_flag.get("description"),
+                default_enabled=None,  # todo
+                type=STANDARD,
+                tags=None,  # todo
+                is_archived=ld_flag.get("archived", False),
+                owners=None,  # todo
+                is_server_key_only=None,  # todo
+            )
+        except IntegrityError:
+            # Feature with this name already exists
```

Comment:
We may actually want to do this check manually - I think that (for example in oracle) we don't always have the db constraint to rely upon.

Reply:
yep 100% agree to check in code.